### PR TITLE
fix: Don't multiplex subscriptions

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/1947

We shouldn't be multiplexing subscriptions as the API returns all subscriptions. So we were basically getting all subscriptions for each subscription, and then using the client subscription id to set the value in the DB. Since we use `ID` as the primary key and do an upsert (at least in v2), the last operation is the one that "took".

~~I think this also means subscription locations should not be multiplexed either and be a relation of subscriptions.
Most of the changes are to support relations in the new SDK list pager API.~~

Removed the relations stuff to simplify things

~~Still a draft as needs some cleanup, which I'll do next week~~

Done

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Test locally on your own infrastructure
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation)) 
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Comment by erezrokah** [comment]
> I think this also means subscription locations should not be multiplexed either and be a relation of subscriptions. Most of the changes are to support relations in the new SDK list pager API.

Not sure about that, maybe since we have the subscriptions already we can make it a top level table

File: plugins/source/azure/resources/services/subscriptions/locations.go [hunk 1]
```diff
@@ -7,18 +7,20 @@ import (
...
 		Name:      "azure_subscriptions_locations",
 		Resolver:  fetchSubscriptionsLocations,
-		Multiplex: client.SubscriptionMultiplex,
+		Multiplex: client.SingleSubscriptionMultiplex,
 		Columns: []schema.Column{
 			{
-				Name:     "subscription_id",
-				Type:     schema.TypeString,
-				Resolver: client.ResolveAzureSubscription,
+				Name:     "subscriptions_subscription_id",
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/subscriptions/locations.go:21]: This is a bit confusing but `ParentIDResolver` get the CQ Id

File: plugins/source/azure/resources/services/subscriptions/subscriptions.go [hunk 2]
```diff
@@ -58,17 +53,27 @@ func Subscriptions() *schema.Table {
 				Type:     schema.TypeString,
 				Resolver: schema.PathResolver("State"),
 			},
+			{
+				Name:     "subscription_id",
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/subscriptions/subscriptions.go:57]: I'll probably revert this and use `ResolveAzureSubscription` with `SingleSubscriptionMultiplex`

File: plugins/source/azure/resources/services/subscriptions/locations.go [hunk 3]
```diff
@@ -53,13 +55,25 @@ func Locations() *schema.Table {
...
+				Name:     "subscription_id",
+				Type:     schema.TypeUUID,
+				Resolver: resolveSubscriptionID,
+			},
 		},
 	}
 }
 
-func fetchSubscriptionsLocations(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+func resolveSubscriptionID(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/subscriptions/locations.go:67]: Not sure why, but getting the subscription ID directly from the location sets a `nil` value in the database.
I'll look into it and probably revert this custom resolver

File: plugins/source/azure/resources/services/subscriptions/subscriptions.go [hunk 4]
```diff
@@ -13,13 +13,8 @@ func Subscriptions() *schema.Table {
 	return &schema.Table{
 		Name:      "azure_subscriptions",
 		Resolver:  fetchSubscriptionsSubscriptions,
-		Multiplex: client.SubscriptionMultiplex,
+		Multiplex: client.SingleSubscriptionMultiplex,
 		Columns: []schema.Column{
-			{
-				Name:     "subscription_id",
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/subscriptions/subscriptions.go:19]: Subscriptions already have the id in `ID` so no need for `subscription_id`

File: plugins/source/azure/resources/services/subscriptions/locations.go (hunk 3 above)
Comment (erezrokah) [reply plugins/source/azure/resources/services/subscriptions/locations.go]: Done

**Comment by erezrokah** [comment]
@hermanschaaf marked this as ready and made you reviewer :) I mad the minimal changes to fix the bug and I think we can do the relations bit if we feel that's the right approach, or required by users.

**Review by hermanschaaf — APPROVED** [review]
@erezrokah Looks good 👍 I couldn't test it locally the other day due to some unrelated local environment issues, but this fixes the bug so let's not block on that--we can always revisit if something still doesn't work right.

**Pull request merged**